### DISCUSSION TITLE
Need to add Auto Increment Id to Custom pivot table entity ?

### DISCUSSION THREAD

**Discussion by huyhoang160593** [mikro-orm/mikro-orm#4992] [Q&A] [answered]
By follow the doc [here](https://mikro-orm.io/docs/collections#custom-pivot-table-entity), I can create a M:N relation between 2 table with a pivot table that have 2 many-to-one properties, but if I want to add a Auto Increment ID to be the third key of the primary key, for now when ever I define those two property to be the primary, the id will lose the auto_increment and cannot be add back. It cost me more than a day without any progress. Can anyone suggest a fix for this ???

**Comment by B4nan** [comment, marked as answer]
Just define a simple PK, there is no need to have a compound key with 3 parts here (in fact, having such a compound key is not supported by the ORM, all parts of composite keys need to have value provided by the user).

You don't even need a custom pivot entity for this, just set `fixedOrder: true` in your m:n relation and it will generate autoincrement PK next to the two FKs.

https://mikro-orm.io/docs/collections#forcing-fixed-order-of-collection-items
